pgx-version-updater/src/main.rs

```diff
@@ -146,90 +160,89 @@ fn main() {
                     .dimmed()
                     .to_string()
                     .as_str(),
             )
         } else {
-            // Bump pacakge version if we can
-            if doc.contains_key("package") {
-                doc["package"]["version"] = value(args.update_version.clone());
+            // Bump package version if we can
+            if let Some(package_version) = doc.get_mut("package").and_then(|p| p.get_mut("version"))
+            {
+                *package_version = value(args.update_version.clone());
             }
         }
 
         // Process dependencies in each file. Generally dependencies can be found in
         // [dependencies], [dependencies.foo], [build-dependencies], [dev-dependencies]
-        for updatable_table in vec!["dependencies", "build-dependencies", "dev-dependencies"] {
-            if doc.contains_table(updatable_table) {
-                let deps_table: &mut Table = doc
-                    .get_mut(updatable_table)
-                    .unwrap()
-                    .as_table_mut()
-                    .unwrap();
-
-                // Attempt to find auto-extracted package names in the various dependency
-                // declarations
-                for package in &updatable_package_names_set {
-                    if deps_table.contains_key(package) {
-                        let dep_item = deps_table.get_mut(package).unwrap();
-
-                        if dep_item.is_table() {
-                            // Tables can contain other tables, and if that's the case we're
-                            // probably at a case of:
-                            //   [dependencies.pgx]
-                            //   version = "1.2.3"
-                            let old_version = dep_item.get("version").unwrap();
-                            let new_version = parse_new_version(
-                                old_version.as_str().unwrap(),
-                                &args.update_version.as_str(),
-                            );
-                            dep_item["version"] = value(new_version);
-                        } else if dep_item.is_inline_table() {
-                            // Inline table covers the case of:
-                            //   [dependencies]
-                            //   pgx = { version = "1.2.3", features = ["..."] }
-                            let inline_table = dep_item.as_inline_table().unwrap();
-
-                            if inline_table.contains_key("version") {
-                                let old_version = inline_table.get("version").unwrap();
-                                let new_version = parse_new_version(
-                                    old_version.as_str().unwrap(),
+        for updatable_table_name in vec!["dependencies", "build-dependencies", "dev-dependencies"] {
+            if let Some(updatable_table) =
+                doc.get_mut(updatable_table_name).and_then(|i| i.as_table_mut())
+            {
```

Comment:
This is pretty good but could be made clearer still, eliminating one more indentation that covers a lot of code.
```suggestion
        for updatable_table in ["dependencies", "build-dependencies", "dev-dependencies"]
        .into_iter()
        .filter_map(|table| doc.get_mut(table).and_then(|item| item.as_table_mut())) {
```

Reply:
We talked briefly about this in Discord, but using `filter_map` here creates some issues:

```
error: captured variable cannot escape `FnMut` closure body
   --> pgx-version-updater/src/main.rs:200:33
    |
173 |         let mut doc = data.parse::<Document>().expect(
    |             ------- variable defined here
...
200 |             .filter_map(|table| doc.get_mut(table).and_then(|item| item.as_table_mut()))
    |                               - ---^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                               | |
    |                               | returns a reference to a captured variable which escapes the closure body
    |                               | variable captured here
    |                               inferred to be a `FnMut` closure
    |
    = note: `FnMut` closures only have access to their captured variables while they are executing...
    = note: ...therefore, they cannot allow references to captured variables to escape
help: consider adding 'move' keyword before the nested closure
    |
200 |             .filter_map(|table| doc.get_mut(table).and_then(move |item| item.as_table_mut()))
    |                                                             ++++

error: could not compile `pgx-version-updater` due to previous error
```

Reply:
Should also be noted that adding `move` as the compiler suggests doesn't seem to help much either:

```
   Compiling pgx-version-updater v0.1.0 (/home/brady/projects/tcdi/pgx/pgx-version-updater)
error: captured variable cannot escape `FnMut` closure body
   --> pgx-version-updater/src/main.rs:200:33
    |
173 |         let mut doc = data.parse::<Document>().expect(
    |             ------- variable defined here
...
200 |             .filter_map(|table| doc.get_mut(table).and_then(move |item| item.as_table_mut()))
    |                               - ---^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                               | |
    |                               | returns a reference to a captured variable which escapes the closure body
    |                               | variable captured here
    |                               inferred to be a `FnMut` closure
    |
    = note: `FnMut` closures only have access to their captured variables while they are executing...
    = note: ...therefore, they cannot allow references to captured variables to escape

error: could not compile `pgx-version-updater` due to previous error
```

Reply:
Oof, I think I was wrong. I forgot about that rule regarding closure lifetimes.